File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left=0;
        int right=height.length-1;
        int maxarea=0;

        while(left<right){
                int h=Math.min(height[right],height[left]);
                int w=right-left;
                int area=h*w;
                maxarea=Math.max(maxarea,area);
            
            if(height[left]<height[right]){
                left++;
            }else{
                right--;
            }
        }
        return maxarea;
    }
}

File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        right=len(height)-1
        left=0
        maxarea=0

        while left<right:
            h=min(height[left],height[right])
            w=right-left
            area=h*w

            if area > maxarea:
                maxarea=area

            if height[left] < height[right]:
                left+=1
            else:
                right-=1
        

        return maxarea

        


            

        